File: clojure/src/sum-even-fibonacci.clj
(defn sum-even-fibonacci [limit]
  (loop [prev 1 curr 2 sum 0] 
    (if (<= curr limit) 
      (let [next (+ prev curr)]
        (if (even? curr) 
          (recur curr next (+ sum curr)) 
          (recur curr next sum))) 
      (sum))))

File: javascript/sum-even-fibonacci.js
const sumEvenFibonacci = (limit) => {
  let prev = 1;
  let curr = 2;
  let sum = 0;
  while (curr <= limit) {
    if (curr % 2 === 0) {
      sum += curr;
    }
    const temp = prev;
    prev = curr;
    curr += temp;
  }
  return sum;
};
